src/ToolBox/SOS/Strike/util.cpp

```diff
@@ -4859,6 +4859,7 @@ HRESULT InitCorDebugInterface()
     hr = g_ExtSymbols->GetNumberModules(&cLoadedModules, &cUnloadedModules);
     if (FAILED(hr))
     {
+        delete pDebuggingImpl;
```

Comment:
These `delete pDebuggingImpl` should not be needed and will probably lead to double frees. The object is automatically freed by the `pClrDebugging` smart pointer.

Reply:
@mikedn Thank you for your comment. I don't see how `pClrDebugging` has influence over `pDebuggingImpl`. Can you elaborate please?

Reply:
The 2 pointers point to the same object. `QueryInterface` basically casts the `this` pointer to the appropriate interface, calls `AddRef` on it and returns it.

Reply:
@mikedn I see now. Any reason why `ToRelease` is being used instead of `unique_ptr`?

Reply:
Hmm, there are probably multiple reasons:
* `unique_ptr` does not work with COM interfaces out of the box
* back when this code was written `unique_ptr` didn't exist
* in general STL is not used in the runtime code (though SOS isn't exactly runtime code)

Reply:
@mikedn Resolved by reverting.